domain/schema/controller.go

```diff
@@ -591,3 +592,83 @@ CREATE TABLE flag (
 );
 `)
 }
+
+func userPermissionSchema() schema.Patch {
+	return schema.MakePatch(`
+CREATE TABLE permission_access_type (
+    id     INT PRIMARY KEY,
+    type   TEXT NOT NULL
+);
+
+CREATE UNIQUE INDEX idx_permission_access_type
+ON permission_access_type (type);
+
+-- Maps to the Access type in core/permission package.
+INSERT INTO permission_access_type VALUES
+    (0, 'read'),
+    (1, 'write'),
+    (2, 'consume'),
+    (3, 'admin'),
+    (4, 'login'),
+    (5, 'addmodel'),
+    (6, 'superuser');
+
+CREATE TABLE permission_object_type (
+    id    INT PRIMARY KEY,
+    type  TEXT NOT NULL
+);
+
+CREATE UNIQUE INDEX idx_permission_object_type
+ON permission_object_type (type);
+
+INSERT INTO permission_object_type VALUES
+    (0, 'controller'),
+    (1, 'cloud'),
+    (2, 'model'),
+    (3, 'offer');
+
+CREATE TABLE permission_object_access (
+    id              INT PRIMARY KEY,
+    access_type     INT NOT NULL,
+    object_type     INT NOT NULL,
+    CONSTRAINT      uc_object_access UNIQUE (access_type, object_type),
+    CONSTRAINT      fk_permission_access_type
+        FOREIGN KEY (access_type)
+        REFERENCES  permission_access_type(id),
```

Comment:
This is causing the UT failure, I have no idea why at this point. Help here is appreciated.